src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs

```diff
@@ -1641,6 +1642,7 @@ public override void VisitInterpolation(IInterpolationOperation operation)
 
             Visit(operation.Expression, "Expression");
             Visit(operation.Alignment, "Alignment");
+            Assert.True(operation.FormatString == null || operation.FormatString.Kind == OperationKind.Literal);
```

Comment:
>Assert.True(operation.FormatString == null || operation.FormatString.Kind == OperationKind.Literal); [](start = 12, length = 100)

When an assert like fails, it will provide no hint what happened in the test log. Please replace it with an ```Assert.Equal``` inside an ```if``` statement, similar how it is done in TestOperationVisitor

Reply:
I am not sure what you mean - assert failure here means `operation.FormatString != null && operation.FormatString.Kind != OperationKind.Literal`. What additional information does your approach give over this one?

---
In reply to: [185968286](https://github.com/dotnet/roslyn/pull/26545#discussion_r185968286) [](ancestors = 185968286)

Reply:
> I am not sure what you mean - assert failure here means operation.FormatString != null && operation.FormatString.Kind != OperationKind.Literal. What additional information does your approach give over this one?

The ```Assert.True``` when fails simply says the value is false when it was expected to be true. The ```Assert.Equal``` tells you what was the actual value when it is not matching the expectation, therefore providing valuable information, which often allows to understand what went wrong without debugging the test. It is fine if you don't want to make this change, I'll do it myself after the PR is merged. 

---
In reply to: [186062344](https://github.com/dotnet/roslyn/pull/26545#discussion_r186062344) [](ancestors = 186062344,185968286)

Reply:
> The Assert.Equal tells you what was the actual value when it is not matching the expectation

Ah, I did not figure this part. Your point is valid, I will fix it.

---
In reply to: [186159571](https://github.com/dotnet/roslyn/pull/26545#discussion_r186159571) [](ancestors = 186159571,186062344,185968286)